test/integration/targets/docker_swarm/tasks/main.yml

```diff
@@ -1,2 +1,9 @@
+- debug:
+    var: docker_api_version
+- debug:
+    msg: '{{ docker_api_version is version("1.35", ">=") }}'
+- debug:
+    msg: 'Conditional failed to work'
+  when: ansible_os_family != 'RedHat' or ansible_distribution_major_version != '6' and docker_api_version is version('1.35', '>=')
 - include_tasks: test_swarm.yml
-  when: ansible_os_family != 'RedHat' or ansible_distribution_major_version != '6'
+  when: ansible_os_family != 'RedHat' or ansible_distribution_major_version != '6' and docker_api_version is version('1.35', '>=')
```

Comment:
The reason why the conditional doesn't do what you want is that you need brackets around the first part, currently it says "either the OS is not Redhat, or the major version is not 6 and the API version is at least 1.35" (which is always true for OpenSUSE and Ubuntu) :)